book/src/pong_tutorial/pong_tutorial_02.md

```diff
@@ -0,0 +1,367 @@
+# Drawing the paddles
+
+Now let's do some drawing! But to draw something, we need something to draw. In 
+Amethyst, those "somethings" are called Entities, which are described by 
+Components.
+
+Amethyst uses Specs for its ECS (Entity-component system), which is a parallel 
+Entity-component system written in Rust. You can learn more about Specs in the 
+[The Specs Book][sb].
+
+## A quick refactor
+
+Let's create a new file called `pong.rs` to hold our core game logic. We can 
+move the `Pong` struct over here, and the `impl State for Pong` block as well. 
+Then, in `main.rs` declare a module:
+
+```rust,ignore
+mod pong; 
+```
+
+And in the `run()` function add:
+
+```rust,ignore
+use pong::Pong;
+```
+
+Now you can just delete various `main.rs` use statements until the Rust compiler 
+stops complaining about unused imports. In `pong.rs` we'll need these use 
+statements to make it through this chapter:
+
+```rust,ignore
+use amethyst::prelude::*;
+use amethyst::ecs::{Component, DenseVecStorage};
+use amethyst::assets::Loader;
+use amethyst::core::cgmath::Vector3;
+use amethyst::core::transform::{LocalTransform, Transform};
+use amethyst::renderer::{Camera, Material, MaterialDefaults, PosTex, MeshHandle, 
+                         Event,KeyboardInput, VirtualKeyCode, WindowEvent};
+```
+
+## Our first Component
+
+In pong.rs let's create our first `Component`, a definition of a paddle.
+
+
+```rust,ignore
+#[derive(PartialEq, Eq)]
+enum Side {
+    Left,
+    Right,
+}
+
+struct Paddle {
+    pub side: Side,
+    pub width: f32,
+    pub height: f32,
+}
+
+impl Paddle {
+    fn new(side: Side) -> Paddle {
+        Paddle {
+            side: side,
+            width: 1.0,
+            height: 1.0,
+        }
+    }
+}
+```
+
+"But that just looks like a regular struct!" you might say. And you're right, 
+here's the special sauce:
+
+```rust,ignore
+impl Component for Paddle {
+    type Storage = DenseVecStorage<Self>;
+}
+```
+
+By implementing `Component` for our `Paddle` struct, and defining the way we'd 
+like that `Component` data stored, we can now add the `Paddle` component to 
+entities in our game. For more on storage types, check out the 
+[Specs documentation][sb-storage].
+
+## Where is the world?
+
+Now it's time to add a new method to our State implementation: `on_start`. 
+Inside this function, we'll `register` our `Paddle` component on the mutable 
+`World` object we're passed by Amethyst's state machine when the game starts up.
+
+```rust,ignore
+fn on_start(&mut self, world: &mut World) {
+  world.register::<Paddle>();
+}
+```
+
+This `World` gets passed around everywhere. It carries with it all the 
+components and resources in our game. Not only the components and resources we 
```

Comment:
Uninitiated reader here would probably go: "Wait what's a resource?"

There's a choice between two improvements I'd make to this:

1. Don't talk about resources yet, because we don't have a resource we want to add to the game.
2. Elaborate on what a resource is and how it contrasts with a component.

Option 2 makes more sense if we consider this to be the "ECS introduction" section, but having an example and a use case might be helpful for introducing resources later which is why I provide option 1.

Reply:
We end up referring to a few resources in this chapter (like the Loader, for instance) so I think I should provide at least a short explanation. The specs documentation is pretty minimal, but from what I understand a resource is a type of data that can be shared between systems and which is independent of entities.